src/components/NotificationDot/NotificationDot.tsx

```diff
@@ -0,0 +1,30 @@
+import React, { cloneElement, Children, ReactElement } from "react";
+import styled from "styled-components";
+import { NotificationDotProps, DotProps } from "./types";
+
+const NotificationDotRoot = styled.span`
+  display: inline-flex;
+  position: relative;
+`;
+
+const Dot = styled.span<DotProps>`
+  display: ${({ invisible }) => (invisible ? "none" : "inline-flex")};
+  position: absolute;
+  top: 0;
+  right: 0;
+  width: 10px;
+  height: 10px;
+  pointer-events: none;
+  border: 1px solid ${({ theme }) => theme.colors.invertedContrast};
+  border-radius: 50%;
+  background-color: ${({ theme }) => theme.colors.failure};
+`;
+
+const NotificationDot: React.FC<NotificationDotProps> = ({ invisible = false, children, ...props }) => (
+  <NotificationDotRoot>
+    {Children.map(children, (child: ReactElement) => cloneElement(child, props))}
```

Comment:
What is the reason for cloning?

Reply:
ButtonMenu sets it's props to it's direct children, so NotificatonDot will receive MenuItems' props then I use clone to pass the props to MenuItems.
https://github.com/pancakeswap/pancake-uikit/blob/master/src/components/ButtonMenu/ButtonMenu.tsx#L14-L23